code/modules/keybindings/bindings_atom.dm

```diff
@@ -41,4 +45,25 @@
 	var/movement_dir = NONE
 	for(var/_key in keys_held)
 		movement_dir |= movement_keys[_key]
-	intended_direction = movement_dir
+	additional_intended_direction = movement_dir
+
+/client/verb/move_key_north()
+	set instant = TRUE
+	set hidden = TRUE
+	intended_direction = NORTH
+
+/client/verb/move_key_south()
+	set instant = TRUE
+	set hidden = TRUE
+	intended_direction = SOUTH
+
+/client/verb/move_key_west()
+	set instant = TRUE
+	set hidden = TRUE
+	intended_direction = WEST
+
+/client/verb/move_key_east()
+	set instant = TRUE
+	set hidden = TRUE
+	intended_direction = EAST
```

Comment:
```suggestion
	intended_direction |= NORTH

/client/verb/move_key_south()
	set instant = TRUE
	set hidden = TRUE
	intended_direction |= SOUTH

/client/verb/move_key_west()
	set instant = TRUE
	set hidden = TRUE
	intended_direction |= WEST

/client/verb/move_key_east()
	set instant = TRUE
	set hidden = TRUE
	intended_direction |= EAST
```

Just to be safe, because we've diagonal movement.

Reply:
This makes no sense, diagonal movement is handled in calculate_move_dir() tracking keystrokes via KeyDown. And more than one +REP verb at the same time from the client does not come from the client

Reply:
Having run a couple of performance tests - the difference in load between this two on live server is likely to be as imperceptible as possible. So, okay, I'll accept the change, why not

Reply:
does it actually make sense behavior wise?

Reply:
> does it actually make sense behavior wise?

Theoretically, if two different +REP commands come from the client in one server tick (which can happen in general, because the client sends these commands more often than the server calls KeyLoop) - such implementation will be useful. However, such cases should be covered by a regular KeyDown as well

Reply:
> This makes no sense, diagonal movement is handled in calculate_move_dir() tracking keystrokes via KeyDown. And more than one +REP verb at the same time from does not come from the client

Well, good to know.